pkg/sql/logictest/tests/cockroach-go-testserver-25.4/cockroach-go-testserver-25_4_test: TestLogic_mixed_version_new_privileges failed

pkg/sql/logictest/tests/cockroach-go-testserver-25.4/cockroach-go-testserver-25_4_test.TestLogic_mixed_version_new_privileges [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21313947?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21313947?buildTab=artifacts#/) on master @ [f206e3fc586dd47b6e4a03b063fe90ec29b1a670](https://github.com/cockroachdb/cockroach/commits/f206e3fc586dd47b6e4a03b063fe90ec29b1a670):


```
=== RUN   TestLogic_mixed_version_new_privileges
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/f3fff1513feed872a29ccdf881c2eeeb/logTestLogic_mixed_version_new_privileges4169791271
    test_log_scope.go:82: use -show-logs to present logs inline
[19:14:12] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/780/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-25.4/cockroach-go-testserver-25_4_test_/cockroach-go-testserver-25_4_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_new_privileges with config cockroach-go-testserver-25.4: 7 tests, 0 failures
    logic.go:4517: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/780/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-25.4/cockroach-go-testserver-25_4_test_/cockroach-go-testserver-25_4_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_new_privileges:121: error while processing
    logic.go:4517: could not reset vectorize mode: dial tcp 127.0.0.1:33837: connect: connection refused
cockroach logs captured in: /artifacts/tmp/_tmp/f3fff1513feed872a29ccdf881c2eeeb/cockroach-logs1898553823
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f3fff1513feed872a29ccdf881c2eeeb/logTestLogic_mixed_version_new_privileges4169791271
--- FAIL: TestLogic_mixed_version_new_privileges (75.68s)
=== RUN   TestLogic_mixed_version_new_privileges/post_upgrade
    logic.go:3537: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/780/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-25.4/cockroach-go-testserver-25_4_test_/cockroach-go-testserver-25_4_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_new_privileges:121: error while processing
    logic.go:3537: os: process already finished
--- FAIL: TestLogic_mixed_version_new_privileges/post_upgrade (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_mixed_version_new_privileges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-62950

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -1134,6 +1134,13 @@ type logicTest struct {
 	skipOnRetry    bool
 	skippedOnRetry bool
 
+	// skippedDueToSnappy is sticky once set: it indicates that the
+	// "Can't find decompressor for snappy" infra flake was detected
+	// (#124966), the testserver cluster has been stopped, and the
+	// remainder of the test should be skipped rather than run further
+	// directives against the dead cluster.
+	skippedDueToSnappy bool
+
 	// declarativeCorpusCollector used to save declarative schema changer state
 	// to disk.
 	declarativeCorpusCollector *corpus.Collector
@@ -1483,6 +1490,8 @@ func (t *logicTest) handleWaitForInitErr(ts testserver.TestServer, err error) {
 			if ts != nil {
 				ts.Stop()
 			}
+			t.skippedDueToSnappy = true
+			t.testserverCluster = nil
 			t.t().Skip("ignoring init did not finish for node error due to snappy error")
 		}
 	}
@@ -2403,6 +2412,9 @@ func (t *logicTest) processTestFile(path string, config logictestbase.TestCluste
 		if *maxErrs > 0 && t.failures >= *maxErrs {
 			break
 		}
+		if t.skippedDueToSnappy {
+			break
+		}
 		// If subtest has no name, then it is not a subtest, so just run the lines
 		// in the overall test. Note that this can only happen in the first subtest.
 		if len(subtest.name) == 0 {
@@ -2422,6 +2434,10 @@ func (t *logicTest) processTestFile(path string, config logictestbase.TestCluste
 			})
 			t.maybeSkipOnRetry(nil)
 		}
+	}
+	if t.skippedDueToSnappy && !t.rootT.Failed() {
+		skip.IgnoreLintf(t.rootT,
+			"skipping remainder of test due to snappy infra flake; see #124966")
 	}
 
 	if (*rewriteResultsInTestfiles || *rewriteSQL) && !t.rootT.Failed() {

